components/dashboard/table/Filters.tsx

```diff
@@ -0,0 +1,130 @@
+import { useFloating } from '@floating-ui/react-dom-interactions';
+import type { RefObject } from 'react';
+import { useEffect, useRef, useState } from 'react';
+
+import type { Issue } from '@/types/components/dashboard/TableTypes';
+
+import Select from '../../form/Select';
+
+interface FilterProps {
+  className?: string;
+  issues: Issue[];
+  allIssues: Issue[];
+  selectedRepo: string;
+  selectedArea: string;
+  setSelectedRepo: (repo: string) => void;
+  setSelectedArea: (area: string) => void;
+}
+
+/**
+ * @description Hook that alerts clicks outside of the passed ref
+ */
+function useOutsideAlerter(ref: RefObject<any>, setOpen: (open: boolean) => void) {
+  useEffect(() => {
+    /**
+     * @description Alert if clicked on outside of element
+     */
+    function handleClickOutside(event: MouseEvent) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        setOpen(false);
+      }
+    }
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [ref]);
```

Comment:
Kindly add a comment explaining this `useEffect`

Reply:
Done